### DISCUSSION TITLE
Non-Newtonian Viscosity

### DISCUSSION THREAD

**Discussion by anicusan** [chaos-polymtl/lethe#318] [Q&A] [answered]
Hello,

Is there any way to set up cases with more complex rheology currently? Perhaps shear-thinning or -thickening, or especially Bingham-type fluids.

If there isn't, how difficult would it be to integrate different equations for the viscosity term in lethe's GLS-Navier-Stokes formulation? If the main work is of the software development type (and not hardcore PDE derivations...) I'd be interested to contribute.

Best wishes,
Leonard

**Comment by blaisb** [comment, marked as answer]
Dear Leonard,
It is actually possible! This is a very new feature and its documentation is not fully fleshed out right now. Since we are currently migrating to a new documentation format, some stuff have been left behind in the wiki. This will be fixed in the current month. @CaroleAnneDaunais  made the development on this part.

To make a long story short. Lethe supports newtonian and non-newtonian flows. The only non-newtonian rheology model supported at the moment is the 4 parameters Carreau-Yasuda model. It is a very generic and flexible one. Adding new rheology model would be a few hours of work because the interface that Carole-Anne has designed is very generic. You just need to add a new rheology model and it will be able to percolate everywhere.

There are no examples at the time but there are a few applications tests:

https://github.com/lethe-cfd/lethe/blob/master/applications_tests/gls_navier_stokes_2d/mms2d_carreau_2nd-order_gls.prm
https://github.com/lethe-cfd/lethe/blob/master/applications_tests/gls_navier_stokes_2d/mms2d_carreau_1st-order_gls.prm
https://github.com/lethe-cfd/lethe/blob/master/applications_tests/gd_navier_stokes_2d/cylinder_carreau_gd.prm

The solver preserves the underlying convergence of the GLS (Qn-Qn) and GD (Qn-Q(n-1)) Schemes.

Feel free to ask around if you need help using it.
